eslint.config.mjs

```diff
@@ -0,0 +1,204 @@
+import { defineConfig } from 'eslint/config'
+import js from '@eslint/js'
+import tseslint from 'typescript-eslint'
+import importPlugin from 'eslint-plugin-import'
+import jsxA11yPlugin from 'eslint-plugin-jsx-a11y'
+import eslintPluginPrettierRecommended from 'eslint-plugin-prettier/recommended'
+
+import reactPlugin from 'eslint-plugin-react'
+import reactHooksPlugin from 'eslint-plugin-react-hooks'
+import simpleImportSortPlugin from 'eslint-plugin-simple-import-sort'
+import sortDestructureKeysPlugin from 'eslint-plugin-sort-destructure-keys'
+import sortReactDependencyArraysPlugin from 'eslint-plugin-sort-react-dependency-arrays'
+import globals from 'globals'
+
+export default defineConfig([
+  {
+    ignores: ['**/node_modules/**', '**/dist/**', '**/build/**', 'src/tools/migrations/steps/steps/template.ts'],
+  },
+  js.configs.recommended,
+  ...tseslint.configs.recommended,
+  reactPlugin.configs.flat.recommended,
+  jsxA11yPlugin.flatConfigs.recommended,
+  {
+    files: ['**/*.{js,jsx,ts,tsx}'],
+    languageOptions: {
+      parser: tseslint.parser,
+      ecmaVersion: 'latest',
+      sourceType: 'module',
+      parserOptions: {
+        ecmaFeatures: {
+          jsx: true,
+        },
+      },
+      globals: {
+        ...globals.browser,
+        ...globals.node,
+        ...globals.commonjs,
+        ...globals.es2021,
+        ...globals.jest,
+        __DEV__: 'readonly',
+      },
+    },
+    plugins: {
+      '@typescript-eslint': tseslint.plugin,
+      react: reactPlugin,
+      'react-hooks': reactHooksPlugin,
+      import: importPlugin,
+      'simple-import-sort': simpleImportSortPlugin,
+      'sort-destructure-keys': sortDestructureKeysPlugin,
+      'sort-react-dependency-arrays': sortReactDependencyArraysPlugin,
+    },
+    settings: {
+      react: {
+        version: 'detect',
+      },
+      'import/resolver': {
+        node: {
+          extensions: ['.js', '.jsx', '.ts', '.tsx'],
+        },
+        webpack: {
+          config: 'webpack.config.babel.js',
+        },
+        typescript: {},
+      },
+    },
+    rules: {
```

Comment:
@yaguzmang in current dev branch I have added this rule:
`'class-methods-use-this': 'off',`
Please add this as well.